packages/zendesk-adapter/e2e_test/adapter.test.ts

```diff
@@ -1144,7 +1167,24 @@ describe('Zendesk adapter E2E', () => {
         .forEach(instanceToAdd => {
           const instance = elements.find(e => e.elemID.isEqual(instanceToAdd.elemID))
           expect(instance).toBeDefined()
-          expect((instance as InstanceElement).value).toMatchObject(instanceToAdd.value)
+          // custom object types have circular references (value and parent)
+          // toMatchObject does not work well with circular references and crashes
+          if ([CUSTOM_OBJECT_TYPE_NAME, CUSTOM_OBJECT_FIELD_TYPE_NAME].includes(instanceToAdd.elemID.typeName)) {
+            const instanceClone = (instance as InstanceElement).clone()
+            const instanceToAddClone = instanceToAdd.clone()
+            const fieldToHandle = instanceClone.elemID.typeName === CUSTOM_OBJECT_TYPE_NAME
+              ? `${CUSTOM_OBJECT_FIELD_TYPE_NAME}s`
+              : CUSTOM_FIELD_OPTIONS_FIELD_NAME
+
+            instanceClone.value[fieldToHandle] = (instanceClone.value[fieldToHandle] ?? [])
+              .map((ref: ReferenceExpression) => ref.value.elemID.getFullName())
```

Comment:
if its a reference expression you don't need the value (same for the next lines as well)
```suggestion
              .map((ref: ReferenceExpression) => ref.elemID.getFullName())
```